File: Day2/Python/main.py
import parts

def main() -> None:
    with open("input.txt", 'r') as f:
        lines = f.readlines()
    
    part1 = parts.Part1( lines )
    part2 = parts.Part2( lines )

    print(f"[Part 1] Total Score : { part1.get_score() }")
    print(f"[Part 2] Total Score : { part2.get_score() }")


if __name__ == "__main__":
    main()

File: Day2/Lua/main.lua

local printf = function(format, vars)
    print(string.format(format, vars))
end

function Main()
    local filename = "input.txt"

    local part1 = require("part1")
    part1:new(filename)
    printf("[Part 1] Total Score : %d", part1:get_score())

    local part2 = require("part2")
    part2:new(filename)
    printf("[Part 2] Total Score : %d", part2:get_score())
end

Main()
